changelog_unreleased/typescript/pr-7144.md

```diff
@@ -0,0 +1,139 @@
+}
+
+type foo7 = /* foo */ (/* bar */) /* baz */ => void
+
+type foo8 = /* foo */ (a: /* bar */ string) /* baz */ => void
+
+let foo9: new /* foo */ (/* bar */) /* baz */ => string;
+
+let foo10: new /* foo */ (a: /* bar */ string) /* baz */ => string;
+
+// Prettier stable
+interface foo1 {
+  bar3 /* foo */(/* baz */); // bat
+  bar /* foo */ /* bar */?(/* baz */) /* bat */;
+  bar2 /* foo */(/* baz */) /* bat */;
+}
+
+interface foo2 {
+  bar /* foo */?(/* bar */ bar: /* baz */ string): /* bat */ string;
+}
+
+interface foo3 {
+  /* foo */ (/* bar */): /* baz */ string;
+}
+
+interface foo4 {
+  /* foo */ (bar: /* bar */ string): /* baz */ string;
+}
+
+interface foo5 {
+  /* foo */ new (/* bar */ a: /* baz */ string): /* bat */ string;
+}
+
+interface foo6 {
+  /* foo */ new (/* baz */): /* bar */ /* bat */ string;
+}
+
+type foo7 = /* foo */ (/* bar */) => /* baz */ void;
+
+type foo8 = /* foo */ (a: /* bar */ string) => /* baz */ void;
+
+let foo9: new (/* bar */) => /* foo */ /* baz */ string;
+
+let foo10: new (/* foo */ a: /* bar */ string) => /* baz */ string;
+
+
+// Prettier master
+interface foo1 {
+  bar3 /* foo */ /* baz */(); // bat
+  bar /* foo */ /* bar */ /* baz */ /* bat */?();
+  bar2 /* foo */ /* baz */() /* bat */;
+}
+
+interface foo2 {
+  bar /* foo */?(/* bar */ bar: /* baz */ string): /* bat */ string;
+}
+
+interface foo3 {
+  /* foo */ (): /* bar */ /* baz */ string;
+}
+
+interface foo4 {
+  /* foo */ (bar: /* bar */ string): /* baz */ string;
+}
+
+interface foo5 {
+  /* foo */ new (/* bar */ a: /* baz */ string): /* bat */ string;
+}
+
+interface foo6 {
+  /* foo */ new (): /* bar */ /* baz */ /* bat */ string;
+}
+
+type foo7 = /* foo */ () => /* bar */ /* baz */ void;
+
+type foo8 = /* foo */ (a: /* bar */ string) => /* baz */ void;
+
+let foo9: new () => /* foo */ /* bar */ /* baz */ string;
+
+let foo10: new (/* foo */ a: /* bar */ string) => /* baz */ string;
+```
+
+```ts
+// Input
+abstract class Test {
+  abstract foo12 /* foo */ (a: /* bar */ string): /* baz */ void
+
+  abstract foo13 /* foo */ (/* bar */) /* baz */
+}
+
+// Prettier stable
+abstract class Test {
+  abstract foo12 /* foo */(a: /* bar */ string): /* baz */ void;
+
+  abstract foo13 /* foo */(/* bar */); /* baz */
+}
+
+// Prettier master
+Error: Comment "foo" was not printed. Please report this error!
+    at https://prettier.io/lib/standalone.js:15543:15
+    at Array.forEach (<anonymous>)
+    at ensureAllCommentsPrinted (https://prettier.io/lib/standalone.js:15541:17)
+    at coreFormat (https://prettier.io/lib/standalone.js:15592:5)
+    at format (https://prettier.io/lib/standalone.js:15832:75)
+    at formatWithCursor (https://prettier.io/lib/standalone.js:15848:14)
+    at https://prettier.io/lib/standalone.js:31794:17
+    at Object.format (https://prettier.io/lib/standalone.js:31802:14)
+    at formatCode (https://prettier.io/worker.js:234:21)
+    at handleMessage (https://prettier.io/worker.js:185:18)
```

Comment:
`stable` is old behavior, `master` is new one

Reply:
ahh, ok i was sure that this is other way around :>

Reply:
fixed